quixstreams/core/stream/stream.py

```diff
@@ -44,23 +58,28 @@ def __init__(
         - "Filter" - to filter values from the Stream.
             The result of a Filter function is interpreted as boolean.
             If it's `True`, the input will be passed downstream.
-            If it's `False`, the `Filtered` exception will be raised to signal that the
-            value is filtered out.
+            If it's `False`, the record will be filtered from the stream.
+        - "Transform" - to transform keys and timestamps along with the values.
+            "Transform" functions may change the keys and should be used with caution.
+            The result of the Transform function is passed downstream to the next
+            functions.
+            If "expand=True" is passed and the function returns an `Iterable`,
+            each item of it will be treated as a separate value downstream.
 
         To execute the functions on the `Stream`, call `.compose()` method, and
         it will return a closure to execute all the functions accumulated in the Stream
         and its parents.
 
         :param func: a function to be called on the stream.
-            It is expected to be wrapped into one of "Apply", "Filter" or "Update" from
-            `quixstreams.core.stream.functions` package.
-            Default - "Apply(lambda v: v)".
+            It is expected to be wrapped into one of "Apply", "Filter", "Update" or
+            "Trasform" from `quixstreams.core.stream.functions` package.
+            Default - "ApplyFunction(lambda value: value)".
         :param parent: a parent `Stream`
         """
         if func is not None and not isinstance(func, StreamFunction):
             raise ValueError("Provided function must be a subclass of StreamFunction")
 
-        self.func = func if func is not None else ApplyFunction(lambda x: x)
+        self.func = func if func is not None else ApplyFunction(lambda value: value)
```

Comment:
As I mentioned to you, wonder if we could consider either:

1. an `IdentityFunction` that you could filter during compose and/or tree,
2. or instead, no default function at all, which to me seems reasonable since it means you haven't added any ops yet.

Reply:
Replied here https://github.com/quixio/quix-streams/pull/375#discussion_r1633578461